src/java/org/apache/cassandra/gms/Gossiper.java

```diff
@@ -158,38 +158,53 @@
 
     private volatile long lastProcessedMessageAt = System.currentTimeMillis();
 
-    //This property and anything that checks it should be removed in 5.0
-    private boolean haveMajorVersion3Nodes = true;
+    // This property and anything that checks it should be removed in 5.0
+    /**
+     * This property is initially set to {@code true} which means that we have no information about the other nodes.
```

Comment:
nit: please merge those 2 comments. 

```suggestion
     /**
     * This property and anything that checks it should be removed in 5.0
     *
     * This property is initially set to {@code true} which means that we have no information about the other nodes.
```

Reply:
ok